reframe/core/schedulers/slurm.py

```diff
@@ -451,11 +454,28 @@ def poll(self, *jobs):
             t_start = time.strftime(
                 '%F', time.localtime(min(job.submit_time for job in jobs))
             )
-            completed = _run_strict(
-                f'sacct -S {t_start} -P '
-                f'-j {",".join(job.jobid for job in jobs)} '
-                f'-o jobid,state,exitcode,end,nodelist'
-            )
+            try:
+                completed = _run_strict(
+                    f'sacct -S {t_start} -P '
+                    f'-j {",".join(job.jobid for job in jobs)} '
+                    f'-o jobid,state,exitcode,end,nodelist'
+                )
+                # Reset the retry counter if the command succeeds
+                self._sacct_retries = 0
+            except SpawnedProcessError as e:
+                if self._sacct_retries < self._max_sacct_retries:
+                    self._sacct_retries += 1
+                    self.log(
+                        (
+                            f'sacct failed with error: '
+                            f'{e.stderr}: have tried {self._sacct_retries} '
+                            f'time(s)'
+                        ),
```

Comment:
No need to parenthesise the string here. Maybe also adapt the message to something like:

```
sacct failed (1/3): <error message>
```